uint16 ch0
uint16 ch1
uint16 ch2
uint16 ch3
uint8 s1
uint8 s2

int16 x
int16 y
int16 z

uint8 l
uint8 r
uint16 key

uint16 wheel